### PR TITLE
Add support for MultipartEncoder

### PR DESCRIPTION
**Pull request by fifty-50**

### CODE REVIEW

**Comment by robertolopezlopez** [comment]
hi @fifty-50 ,

Thank you for taking your time creating this PR. Could you please add any unit tests?

Regards,

Roberto

**Review by robertolopezlopez — CHANGES_REQUESTED** [review]
Please, add unit tests

**Comment by fifty-50** [comment]
Hi @robertolopezlopez, I have added some tests.

**Review by robertolopezlopez — CHANGES_REQUESTED** [review]

**Comment by robertolopezlopez** [comment]
I have an extra question @fifty-50 ,

Which is the endpoint that will benefit from this PR? Which is the intended use case? Because we will need to test manually as well before merge :-)

Regards,

Roberto

File: akamai/edgegrid/test/test_edgegrid.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import os
 import re
 import requests
+import requests_toolbelt
```
[the remark this answers is not in the document]
Comment (fifty-50) [reply akamai/edgegrid/test/test_edgegrid.py:30]: added it

**Comment by fifty-50** [comment]
Hi @robertolopezlopez, the [/case-management/v3/cases/{caseId}/attachments](https://techdocs.akamai.com/case-mgmt/reference/post-attachments) endpoint accepts multipart request payloads, which can be large in size. Adding support for MultipartEncoder would allow streaming of such large payloads.

HTTPie uses MultipartEncoder for all multipart requests, and so it fails when trying to send any multipart request using EdgeGrid authentication. If MultipartEncoder support is added in the edgegrid-python library, the HTTPie integration will also be able to sign and send multipart requests.

**Comment by robertolopezlopez** [comment]
Hi @fifty-50 , if you could help me manually test this change, we may merge this branch quickly. Please ping me through webex

**Review by robertolopezlopez — APPROVED** [review]
Hi @fifty-50 ,

Looks great to me , thank you very much for this PR. We will try to release this change and upgrade the dependency in httpie-edgegrid as soon as we can.

Regards,

Roberto

**Review by kaush- — APPROVED** [review]

**Pull request merged**